File: generate-api-rest-front/src/app/generate/interfaces/attributeModel.ts
import { AttributeFKModel } from "./attributeFKModels";
import { AttributeNormalModel } from "./attributeNormalModels";

export interface AttributeModel {
    typePK:                string;
    namePK:                string;
    isSerial:              boolean;
    nameColumnDB:          string;
    attributeFKModels?:     AttributeFKModel[];
    attributeNormalModels?: AttributeNormalModel[];
}

File: generate-api-rest/src/main/java/mx/unach/generateapirest/models/AttributeModel.java
package mx.unach.generateapirest.models;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Builder
public class AttributeModel {
  
  @Getter @Setter
  private String typePK;

  @Getter @Setter
  private String namePK;

  @Getter @Setter
  private String nameColumnDB;

  @Getter @Setter
  private Boolean isSerial;

  @Getter @Setter
  private List<AttributeFKModel> attributeFKModels;

  @Getter @Setter
  private List<AttributeNormalModel> attributeNormalModels;

}
